hazelcast/src/test/java/com/hazelcast/test/AbstractHazelcastClassRunner.java

```diff
@@ -122,8 +126,24 @@ private static void initialize() {
         System.setProperty("hazelcast.wait.seconds.before.join", "1");
         System.setProperty("hazelcast.local.localAddress", "127.0.0.1");
         System.setProperty("java.net.preferIPv4Stack", "true");
-    }
+        //override default async executor of hazelcast so that it can report correct test names in test runs
+        //if ForkJoinPool parallelism is less than 1, `thread-per-task` will be used.
+        //In that case there is no need to override defaultAsyncExecutor
+        if (ForkJoinPool.getCommonPoolParallelism() > 1) {
+            try {
+                Field field = ConcurrencyUtil.class.getDeclaredField("DEFAULT_ASYNC_EXECUTOR");
+                field.setAccessible(true);
 
+                Field modifiersField = Field.class.getDeclaredField("modifiers");
+                modifiersField.setAccessible(true);
+                modifiersField.setInt(field, field.getModifiers() & ~Modifier.FINAL);
+
+                field.set(null, new TestLoggingUtils.CustomTestNameAwareForkJoinPool());
+            } catch (NoSuchFieldException | IllegalAccessException e) {
+                throw new RuntimeException("Could not initialize defaultAsyncExecutor", e);
```

Comment:
Should test execution fail if we cannot replace the default async executor?

Reply:
Since I am dealing with a reflection, I felt safer to fail-fast. 
If there is a JDK where this will not work, we can see it on pr builders and address it. 
If I just log and continue, we may never know. 
Does that make sense ?